File: simplex.java/src/main/java/kr/co/tmon/simplex/actions/IParameterizedActionBinder.java
package kr.co.tmon.simplex.actions;

/**
 * 파라미터를 전달용 액션의 래핑 인터페이스
 * @author yookjy
 *
 * @param <TAction> 액션 타입
 * @param <TParam> 액션에 전달할 파라미터 타입
 * @param <TResult> 액션의 결과 타입
 */
public interface IParameterizedActionBinder<TAction extends IParameterizedAction<TParam, TResult>, TParam, TResult>
	extends IActionBinder<TAction, TResult> {
}

File: Simplex.kotlin/src/main/kotlin/kr/co/tmon/simplex/actions/IParameterizedActionBinder.kt
package kr.co.tmon.simplex.actions

/**
 * 액션 래핑 인터페이스
 * @param <TAction> 액션 타입
 * @param <TParam> 액션의 매개변수 타입
 * @param <TResult> 액션의 결과값 반환 타입
 */
interface IParameterizedActionBinder<TAction : IParameterizedAction<TParam, TResult>, TParam, TResult>
    : IActionBinder<TAction, TResult>